Integrations/integration-QRadar.yml

```diff
@@ -42,296 +38,554 @@ configuration:
+        if isinstance(d1, list):
+            return map(lambda x: filter_dict_nonintersection_key_to_value(x, d2), d1)
+        elif isinstance(d1, dict) and isinstance(d2, dict):
+            d2values = d2.values()
+            return dict((k,v) for k,v in d1.items() if k in d2values)
+        return d1
+
+    def dict_values_to_comma_seperated_string(dic):
+        return ','.join(str(v) for v in dic.itervalues())
+
+    def get_entry_for_object(title, obj, headers=None, contextKey=None):
+        if len(obj) == 0:
+            return "There is no output result"
+        obj = filter_dict_null(obj)
+        if headers and isinstance(obj, dict):
+            headers = list(set(headers).intersection(set(obj.keys())))
+        ec = { contextKey: obj } if contextKey else obj
         return {
-            "name": incident.id + ' ' + incident.description,
-            "labels": labels,
-            "rawJSON": JSON.stringify(incident),
-            "occurred": occured
-        }
-    }
-
-    var sendRequest = function(method, url, headers, queryName, body) {
-        var httpParams = {
-                    Headers: headers,
-                    Method: method,
-                    Body: body,
-                };
-        if (token) {
-            if (!httpParams.Headers) {
-                httpParams.Headers = {};
-            }
-            httpParams.Headers['SEC'] = [token];
-        } else {
-            httpParams.Username = username;
-            httpParams.Password = password;
-        }
-        var res = http(
-                url,
-                httpParams,
-                insecure,
-                params.proxy
-            );
-        if (res.StatusCode < 200 || res.StatusCode >= 300) {
-            throw 'Failed to ' + queryName + ', request status code: ' + res.StatusCode + ' and Body: ' + res.Body + '.';
+            'Type': entryTypes['note'],
+            'Contents': obj,
+            'ContentsFormat': formats['json'],
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': tableToMarkdown(title, obj, headers),
+            'EntryContext': ec
         }
-        try {
-          return JSON.parse(res.Body);
-        } catch (err) {
-            logInfo('QRadar error: Could stringify respons for ' + url + '. Error is ' + err + '. Returned body is ' + res.Body);
-            throw 'QRadar error\n Could stringify respons for ' + url + '\n Error is ' + err + '\n Returned body is ' + res.Body;
-        }
-    }
 
-    var getOffenses = function(filter, fields, range) {
-        var urlArgs = {};
-        if (filter) {
-            urlArgs.filter = filter;
-        }
-        if (fields) {
-            urlArgs.fields = fields;
-        }
-        var headers = {'Content-Type': ['application/json']};
-        if (range) {
-            headers.Range = ['items=' + range];
+    def get_entry_for_assets(title, obj, human_readable_obj, headers=None):
```

Comment:
if you create helper method for specific command then better write this function near that command

Reply:
This will break the structure used for the script, where all the utilities funcs are together, but I will do so and also add comments to each method to make it clearer